src/main/java/de/ait/secondlife/controllers/OfferController.java

```diff
@@ -0,0 +1,211 @@
+import de.ait.secondlife.exception_handling.OfferExceptionHandler;
+import de.ait.secondlife.exception_handling.exeptions.PaginationParameterIsWrongException;
+import de.ait.secondlife.services.interfaces.OfferService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/v1/offer")
+@RequiredArgsConstructor
+@Tag(name = "Offer controller", description = "Controller for some operations with available offer")
+@ApiResponses(value = {
+        @ApiResponse(responseCode = "400", description = "Bad request", content = @Content(
+                mediaType = "application/json",
+                schema = @Schema(implementation = ResponseMessageDto.class)
+        )),
+        @ApiResponse(responseCode = "404", description = "Resource not found", content = @Content(
+                mediaType = "application/json",
+                schema = @Schema(implementation = ResponseMessageDto.class)
+        )),
+        @ApiResponse(responseCode = "500", description = "Internal server error", content = @Content(
+                mediaType = "application/json",
+                schema = @Schema(implementation = ResponseMessageDto.class)
+        ))
+})
+public class OfferController implements OfferExceptionHandler {
+
+    private final OfferService service;
+
+    @GetMapping("/all")
+    @Operation(
+            summary = "Get all offer ",
+            description = "Receiving all offer available in the database with pagination"
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = OfferResponseWithPaginationDto.class))),
+    })
+    public ResponseEntity<OfferResponseWithPaginationDto> getAll(
+            @RequestParam(defaultValue = "0")
+            @Parameter(description = "Requested page number. ", example = "0")
+            int page,
+            @RequestParam(defaultValue = "10")
+            @Parameter(description = "Number of entities per page. ", example = "0")
+            int size,
+            @RequestParam(defaultValue = "createdAt")
+            @Parameter(description = "Scoring field. ", example = "createdAt")
```

Comment:
Is this "Sorting field" or something else?

Reply:
field by which sorting is performed